File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/job/meta/PrepareJobDownloadResponse.scala
package com.webank.eggroll.core.deepspeed.job.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.containers.meta.ContainerContent
import com.webank.eggroll.core.meta.Deepspeed
import com.webank.eggroll.core.meta.DeepspeedDownload.{PrepareDownloadRequest, PrepareDownloadResponse}

import scala.language.implicitConversions

case class PrepareJobDownloadResponse(sessionId: String, content :String)

object PrepareJobDownloadResponse {

  implicit def deserialize(byteString: ByteString): PrepareJobDownloadResponse = {
    val proto = PrepareDownloadResponse.parseFrom(byteString)
    PrepareJobDownloadResponse(
      sessionId = proto.getSessionId,
      content = proto.getContent)
  }

  implicit def serialize(src: PrepareJobDownloadResponse): Array[Byte] = {
    PrepareDownloadResponse.newBuilder()
      .setSessionId(src.sessionId)
      .setContent(src.content).build()
      .toByteArray
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/PrepareJobDownloadResponse.java
package com.eggroll.core.pojo;

import com.google.protobuf.InvalidProtocolBufferException;
import com.webank.eggroll.core.meta.DeepspeedDownload;
import lombok.Data;

@Data
public class PrepareJobDownloadResponse implements RpcMessage {

    String sessionId;
    String content;

    @Override
    public byte[] serialize() {
        return DeepspeedDownload.PrepareDownloadResponse.newBuilder().setContent(content).setSessionId(sessionId).build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {
        try {
            DeepspeedDownload.PrepareDownloadResponse prepareDownloadResponse = DeepspeedDownload.PrepareDownloadResponse.parseFrom(data);
            this.sessionId = prepareDownloadResponse.getSessionId();
            this.content = prepareDownloadResponse.getContent();
        } catch (InvalidProtocolBufferException e) {
            e.printStackTrace();
        }
    }
}
